IoTuring/Entity/Deployments/Wifi/Wifi.py

```diff
@@ -0,0 +1,174 @@
+        self.specificGetWireless = self.getWirelessMethods[self.platform]
+
+        self.configuredNic = self.GetFromConfigurations(CONFIG_KEY_WIFI)
+
+    def InitWindows(self):
+        print("InitWindows")
+
+    def InitLinux(self):
+        
+
+
+    def InitMac(self):
+        print("InitMac")
+
+    def UpdateWindows(self):
+        print("UpdateWindows")
+
+    def UpdateLinux(self):
+        p = self.RunCommand(["iwconfig", self.configuredNic])
+        wifi_info = {}
+        
+        for key, pattern in self.patterns.items():
+            match = re.search(pattern, p.stdout)
+            if match:
+                wifi_info[key] = match.group(1) # storing in the dict maybe redundant, but better readable
+                
+        self.SetEntitySensorValue(
+            key=KEY_SIGNAL_STRENGTH,
+            value=wifi_info[wifi_info[]]
+        )
+
+    def UpdateMac(self):
+        print("UpdateMac")
+
+    def GetWirelessStrength_Linux(self):
+        p = self.RunCommand(["iwconfig", self.configuredNic])
+        if p.stderr:
+            raise Exception("error during iwconfig")
+        elif p.stdout:
+            signalStrength = re.findall(f"Signal level=(-[0-9]+) dBm", p.stdout)
+        return signalStrength[0]
+
+    def GetWirelessStrength_Windows(self):
+        p = subprocess.Popen(
+            "netsh wlan show interfaces", stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        out = p.stdout.read().decode(errors="ignore")
+        if "Segnale" in out:  # Italian support
+            m = re.findall(
+                "Nome.*?:.*? ([A-z0-9 \-]*).*?Segnale.*?:.*?([0-9]*)%", out, re.DOTALL
+            )
+        elif "Signal" in out:  # English support
+            m = re.findall(
+                "Nome.*?:.*?([A-z0-9 ]*).*?Signal.*?:.*?([0-9]*)%", out, re.DOTALL
+            )
+        else:
+            self.Log(
+                self.Logger.LOG_ERROR,
+                "Can't get signal strength data in your region. Please open a Git issue with and show the output of this command in the CMD: 'netsh wlan show interfaces' ",
+            )
+            self.supports_signal_strength = False
+            # TODO Remove the signal strength topic for all the inets here !
+        p.communicate()
+        return m
+    def GetWirelessStrength_Mac(self):
+        raise NotImplementedError
+    
+    def Update(self):
+        self.specificUpdate()
+
+    @classmethod
+    def ConfigurationPreset(cls) -> MenuPreset:
+        # Get the choices for menu:
+        NIC_CHOICES = []
+
+        nics = psutil.net_if_addrs()
+
+        for nic in nics:
+            nicinfo = nics[nic]
+            # try to get nic ip
+            nicip4 = ""
+            nicip6 = ""
+            nicmac = ""
+            p = cls.RunCommand(["iwconfig", nic]) # type hint return missing in subprocess.CompletedProcess
+            if not bool(p.stdout): # if there is nothing in stdout for this nic it is no wireless interface
```

Comment:
How would i run .RunCommand inside the configuration? If i understand correctly this class is not fully initialized

Reply:
Yes, you are right you cannot call that here. But I see you found you can run `OsD.CommandExists` here instead.

Reply:
Problem is: i'd like to run iwconfig and the others here to only show proven to be wireless network cards.